python-threatexchange/threatexchange/exchanges/signal_exchange_api.py

```diff
@@ -277,3 +314,148 @@ def naive_convert_to_signal_type(
                 ret[s_type] = inner
             inner[signal_str] = metadata
         return ret
+
+
+class SignalExchangeAPIInvalidAuthException(Exception):
+    """
+    An exception you can use to hint users their authentification is bad
+
+    This can be because it's incorrectly formatted, it's been expired,
+    or a multitude of other reasons.
+    """
+
+    def __init__(self, src_api: TSignalExchangeAPICls, message: str) -> None:
+        self.src_api = src_api
+        self.message = message
+
+
+class SignalExchangeAPIMissingAuthException(Exception):
+    """
+    An exception you can use to hint users how to authentificate your API
+    """
+
+    def __init__(
+        self, src_api: TSignalExchangeAPICls, *, file_hint: str = "", env_hint: str = ""
+    ) -> None:
+        self.src_api = src_api
+        self.hints: t.List[str] = []
+        if env_hint:
+            self.add_env_hint(env_hint)
+        if file_hint:
+            self.add_file_hint(file_hint)
+
+    def add_file_hint(self, filename: str) -> None:
+        self.hints.append(
+            f"creating (and maybe `chmod 400`) a file with credentials at {filename}"
+        )
+
+    def add_env_hint(self, variable_name: str) -> None:
+        self.hints.append(f"populating an environment variable called {variable_name}")
+
+    def pretty_str(self) -> str:
+        lines = [
+            f"Couldn't authenticate {self.src_api.get_name()}, "
+            "it's missing authentification!"
+        ]
+        if self.hints:
+            lines.append("You can fix this by:")
+            for i, hint in enumerate(self.hints, 1):
+                lines.append(f"  {i}. {hint}")
+        return "\n".join(lines)
+
+
+CredentialSelf = t.TypeVar("CredentialSelf", bound="CredentialHelper")
+
+
+class CredentialHelper:
```

Comment:
Here's the majority of the new stuff.

The long term goal here is that we can use something like

```
for_collab(cls: t.Type[Self], collab: CollatT, auth: AuthT = None) -> t.Self:
```
And use reflection on AuthT to auto-generate options on the CLI.

This PR was already a nightmare to produce, and I want to avoid any stabs at fixing anything other than the interface on this pass itself.

Adding this arg won't be backwards compatible though, and the alternative was using Schatten's favorite: global state!